docs/30DaysOfPWA/advanced-capabilities/03.md

```diff
@@ -1,15 +1,123 @@
+
+And this is how it looks on macOS:
+
+![Screenshot of the sharing UI in Microsoft Edge on macOS](_media/day-03/sharing-mac.jpg)
+
+As shown in the images above, the action of sharing invokes the operating system's share dialog. This creates a seamless integrated experience independent of where you are sharing from, allowing the user to send information between applications. Now, here is where it gets really interesting.
+Now that I have laid out the concept of sharing between apps, let's bring PWAs into the mix. We can use the `Web Share API` and `Web Share Target API` to invoke this sharing dialog and share specific content, allowing your web app to behave like any other installed app.
+
+## Our demo PWA
+To demonstrate how the `Web Share API` works, I will use the PWinter PWA, available [here](https://aka.ms/learn-PWA/30Days-2.3/github.com/diekus/pwinter). This PWA allows a user to create custom themed PWA logos. These logos can then be saved as SVG files to be shared.
+
+![Screenshot of a PWA that prints PWA logos named "PWinter"](_media/day-03/pwinter.jpg)
+
+## Sharing from your application
+Let’s start by looking at how to share from your web application. We want to be able to share data from the PWA to another app that can handle the type of data being shared. As is the case for many things web, “there’s an API for that." For this specific scenario, the Web Share API comes to the rescue, and we can use the `navigator.share()` method, which takes as a parameter the data we want to share. This data can be composed of a url, text, title, and even an array of files, if the system supports it. This will return a [Promise](https://developer.mozilla.org/docs/Web/JavaScript/Guide/Using_promises) that resolves if the data is shared. As is the case with advanced powerful APIs, you need to be [running under HTTPS](https://microsoft.github.io/win-student-devs/#/30DaysOfPWA/core-concepts/02?id=_1-https) to use the `Web Share API`.
+
+### Sharing a link
+The PWinter app can share its URL, making it easy to distribute the address to the app with another user. In order to do so, it displays an icon on the lower right corner that when clicked, invokes the Web Share API. The following code shares a title, text and url via the operating system’s share dialog.
+
+```javascript
+// Shares a link with the Web Share API
+async function shareLogo() {
+    let fileToShare = createFileForSharing(preparePWALogoforSVG());
+    let filesArray = [];
+    filesArray[0] = fileToShare;
+    if (navigator.canShare && navigator.canShare({ files: filesArray })) {
+        await navigator.share({
+            files: filesArray,
+            title: 'My PWA Logo',
+            text: 'Custom PWA logo from The PWinter.'
+        })
+        .then(() => console.log('Share was successful.'))
+        .catch((error) => console.log('Sharing failed', error));
+        }
+        else {
+            console.log(`System doesn't support sharing.`);
+    }
+};
+```
```

Comment:
```suggestion
```javascript
// Shares a link with the Web Share API
async function shareLogo() {
    let fileToShare = createFileForSharing(preparePWALogoforSVG());
    let filesArray = [];
    filesArray[0] = fileToShare;
    if (navigator.canShare && navigator.canShare({ files: filesArray })) {
        try {
            await navigator.share({
                files: filesArray,
                title: 'My PWA Logo',
                text: 'Custom PWA logo from The PWinter.'
            });
            console.log('Share was successful.');
        } catch (error) {
            console.log('Sharing failed', error);
        }
    } else {
        console.log(`System doesn't support sharing.`);
    }
};
```
```

Reply:
@aarongustafson @diekus How does this look?